snow/consensus/snowball/consensus.go

```diff
@@ -77,23 +70,17 @@ type Nnary interface {
 }
 
 // Binary is a snow instance deciding between two values.
-// The caller samples k nodes and then calls
-// 1. RecordSuccessfulPoll if choice collects >= alphaConfidence votes
-// 2. RecordPollPreference if choice collects >= alphaPreference votes
-// 3. RecordUnsuccessfulPoll otherwise
+// The caller samples k nodes and calls RecordPoll with the result.
+// RecordUnsuccessfulPoll resets the confidence counters when one or
+// more consecutive polls fail to reach alphaPreference votes.
 type Binary interface {
 	fmt.Stringer
 
 	// Returns the currently preferred choice to be finalized
 	Preference() int
 
-	// RecordSuccessfulPoll records a successful poll towards finalizing the
-	// specified choice
-	RecordSuccessfulPoll(choice int)
-
-	// RecordPollPreference records a poll that preferred the specified choice
-	// but did not contribute towards finalizing the specified choice
-	RecordPollPreference(choice int)
+	// RecordPoll records the results of a network poll
+	RecordPoll(count, choice int)
```

Comment:
This is pretty nit-picky but I think switching the order of the params here looks more natural (as w/ the rest of the RecordPoll signatures we change in this PR)

Reply:
Ya you're right, this also matches the return value of `bag.Mode()` https://github.com/ava-labs/avalanchego/blob/41c12018063d4ef730e78af3fa04ad6a1997db70/snow/consensus/snowball/flat.go#L30.

Will update

Reply:
Hmm further thought looking through this, I'd say I actually prefer this as is because the `Unary` interface only takes in the `count` parameter, so it seems slightly more natural to keep that as the first parameter for `Binary` and `Nnary` as well. I think this was my original rationale for choosing this order iirc, but will leave this as is.